winml/lib/Api.Image/DeviceHelpers.cpp

```diff
@@ -133,6 +133,9 @@ static HRESULT IsFloat16Blocked(ID3D12Device& device, bool* isBlocked) {
 }
 
 bool IsFloat16Supported(const winrt::Windows::AI::MachineLearning::LearningModelDevice& device) {
+  if (device.AdapterId().HighPart == 0 && device.AdapterId().LowPart == 0) {
```

Comment:
If I remember correctly, there's an internal implementation of LearningModelDevice you can call.

device.as<winrt::Windows::AI::MachineLearning::implementation::LearningModelDevice>()
or device.as<winmlp::LearningModelDevice>()

and then you can call IsCpuDevice()

Reply:
Updated!